### DISCUSSION TITLE
User dynamic stylesheet (theme

### DISCUSSION THREAD

**Discussion by clinvest** [filamentphp/filament#11380] [Help] [answered]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

Hi guys,

How would you to load a "user theme stylesheet" for the filament admin ?

I have been doing some tests and researches but I don't find the clean way to do it. For now my tests are: 

```
->renderHook(
    PanelsRenderHook::STYLES_AFTER,
    function() {
        $user = Auth::user();
        $theme = $user->account?->slug;
        return '<style>' . Vite::content('resources/themes/' . $theme . '/' . $theme . '.css') . '</style>';
        // return '<link href="' . asset('themes/' . $theme . '/theme.css') . '" rel="stylesheet">';
        // return Blade::render('<span class="m-auto">test</span>');
    }
);
```

But I cannot do this way, I will have to use "important" everywhere. I need to use the filament way but how to dynamically register the filament theme? Any ideas?

**Comment by clinvest** [comment]
Another idea would be to use getExtraBodyAttributes to add a "theme-name" to the body class. Is there a way to do it for all the panel and not only on a ressource?

**Reply by leandrocfe** [reply, marked as answer]
Create a middleware and add it in the panel provider

```php
$panel
...
->middleware([
...
    YourMiddleware::class,
])
```

YourMiddleware.php
```php
public function handle(Request $request, Closure $next): Response
{
    $user = auth()->user();
    $theme = $user->account?->slug;
    $panel = filament()->getCurrentPanel();
    $panel->viteTheme('resources/css/filament/'.$theme.'.css');

    return $next($request);
}
```

**Reply by clinvest** [reply]
Thank you very much Leandro I will try that :)
